### PR TITLE
istioctl: add  example about filter pod by deployment name for proxy-config

### PR DESCRIPTION
**Pull request by my-git9**
**Please provide a description of this PR:**

Support for filtering Pods by label selector to `istioctl pc`.

Related discussions:  https://github.com/istio/istio/pull/53089#issuecomment-2345082433

### CODE REVIEW

File: istioctl/pkg/proxyconfig/proxyconfig.go [hunk 1]
```diff
@@ -325,6 +325,9 @@ func clusterConfigCmd(ctx cli.Context) *cobra.Command {
 		Example: `  # Retrieve summary about cluster configuration for a given pod from Envoy.
   istioctl proxy-config clusters <pod-name[.namespace]>
 
+  #  Retrieve summary about cluster configuration for a pod under a deployment
```
Comment (hanxiaop) [inline istioctl/pkg/proxyconfig/proxyconfig.go:328]: Can you make the examples more consistent? I see some sentences include `.` while others do not, and some have `from envoy` and some don't.

File: releasenotes/notes/53145.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+releaseNotes:
+  - |
+    **Added** support for filtering Pods by Deployment name to `istioctl pc`.
```
Comment (hanxiaop) [inline releasenotes/notes/53145.yaml:6]:
```suggestion
    **Added** support for filtering Pods by Deployment name to `istioctl proxy-config`.
```

**Comment by my-git9** [comment]
It seems that we just need to add an example: https://github.com/istio/istio/pull/53089#issuecomment-2352939055

File: istioctl/pkg/proxyconfig/proxyconfig.go [hunk 3]
```diff
@@ -397,6 +400,9 @@ func allConfigCmd(ctx cli.Context) *cobra.Command {
 		Example: `  # Retrieve summary about all configuration for a given pod from Envoy.
   istioctl proxy-config all <pod-name[.namespace]>
 
+  # Retrieve summary about all configuration for a pod under a deployment
+  istioctl proxy-config all deployment/<deployment-name[.namespace]> from Envoy.
+
```
Comment (MorrisLaw) [inline istioctl/pkg/proxyconfig/proxyconfig.go:405]: nit: sentence should be ordered similar to the other comments you added:
```suggestion
  # Retrieve summary about all configuration for a pod under a deployment from Envoy.
  istioctl proxy-config all deployment/<deployment-name[.namespace]>
```

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**